cli/src/app.rs

```diff
@@ -347,7 +347,49 @@ pub fn add_subcommands(command: Command) -> Command {
                             .help("Name for the new group")
                             .required(true),
                     ),
-                ),
+                )
+                .subcommand(
+                    Command::new("members").about("List group members").args(&[
+                        Arg::new("group")
+                            .short('g')
+                            .long("group")
+                            .value_name("GROUP")
+                            .help("Group to list the members for")
+                            .required(true),
```

Comment:
This might be personal preference, but I think required arguments are more natural when they are positional. So I much prefer this:

```
phylum group members list <GROUP_NAME>
phylum group members add <GROUP_NAME> <EMAIL>
phylum group members remove <GROUP_NAME> <EMAIL>
```

Over this:

```
phylum group members -g <GROUP_NAME> list
phylum group members -g <GROUP_NAME> add <EMAIL>
phylum group members -g <GROUP_NAME> remove <EMAIL>
```

Reply:
> With `--group` it's a little easier since all positional arguments can be swapped since they're all just user emails

That's fair. You have convinced me.

> The way it's currently done the `--group` location is clear and enforced by clap, leaving no room for screwups.

I'm not sure this is a good thing. I think it makes sense, for example, to allow `phylum group members add --group my-group kyle@phylum.io` in addition to `phylum group members --group my-group add kyle@phylum.io`. And clap has an easy way to do exactly that with [`Arg::global`](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.global).

Reply:
We've run into consistency issues with `Arg::global` before. I **really** don't think it's as good as a solution as you make it out to be.

All `group member` options require a specific group, so enforcing it at **ONE** location makes it absolutely unambiguous which group is operated upon.

You say that it makes sense to allow both `member add --group` and `member --group add`, but you give no reason for why you think this is beneficial. If you ask me, this just adds unnecessary ambiguity. The user needs to specify it either way and with the way it's done right now, that's absolutely clear. Having it both way will just introduce potential for confusion.

Reply:
> We've run into consistency issues with `Arg::global` before. I really don't think it's as good as a solution as you make it out to be.

This is referring to #749? The issue was simply that we had an argument with the same name at two levels, causing overlap once we marked the higher level flag as global. That issue was simple to fix and does not arise in this case.

> You say that it makes sense to allow both `member add --group` and `member --group add`, but you give no reason for why you think this is beneficial. If you ask me, this just adds unnecessary ambiguity. The user needs to specify it either way and with the way it's done right now, that's absolutely clear. Having it both way will just introduce potential for confusion.

`phylum group member add --group my-group kyle@phylum.io` is not ambiguous. It is clear what the command will do. I don't see how that could confuse someone. On the contrary, if they remember that there is a `phylum group member add` command, it's probably more natural that they would type it this way.

Reply:
> This is referring to https://github.com/phylum-dev/cli/pull/749? The issue was simply that we had an argument with the same name at two levels, causing overlap once we marked the higher level flag as global. That issue was simple to fix and does not arise in this case.

I don't remember the exact issues, but wasn't there a problem where global flags could still be specified in multiple locations but then only one of them would actually work? I recall it not being particularly intuitive.

> phylum group member add --group my-group kyle@phylum.io is not ambiguous. It is clear what the command will do. I don't see how that could confuse someone.

The ambiguity here is not what the command will do, but what the "correct" way to do things are. Having three different ways to do the same thing with no good reason for it isn't an advantage for a CLI, it's a massive disadvantage. It confuses users and makes correct usage unclear and will seed doubt in the CLI actually doing what it's supposed to.

If you give a user **one** correct way to do things and that way is always how they're forced to do things then things are clear and make sense, thus giving people trust in the stuff actually working like it's supposed to. It's one of the big complaints people have about `git`.

Reply:
> I recall it not being particularly intuitive.

That is true of most things in `clap`, so you are probably correct here.

> If you give a user **one** correct way to do things and that way is always how they're forced to do things then things are clear and make sense, thus giving people trust in the stuff actually working like it's supposed to.

This is generally true. But it can be taken too far. For example, we allow all of these similar commands:

```
phylum group member --group x list
phylum group member --group=x list
phylum group member -g x list
phylum group member -g=x list
```

Which one is "correct"? I'm not sure. If you ask `phylum group member list --help`, it says the first one:

```
❯ phylum group member list --help
List group members

Usage: phylum group member --group <GROUP> list [OPTIONS]
[ -- SNIP -- ]
```

But it definitely be too restrictive to disallow the others just so that we can pick one correct way. How is this any different?

Reply:
> This is generally true. But it can be taken too far. For example, we allow all of these similar commands:

The examples listed here have nothing to do with our CLI though. They're just commonly known aliases that terminal users should generally be aware of. No matter how our CLI behaves, users should know that `--group=x` and `--group x` would be the same even without using our CLI.

Reply:
After further discussion, we discovered that [`Arg::global`](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.global) cannot be used in conjunction with [`Arg::required`](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.required). Other methods of making this work would almost certainly result in either unmaintainable code or inconsistent and/or confusing error messages.

That makes my suggestion impractical and means that we should continue with the code as written.